### PR TITLE
Corosync pod watcher

### LINKED ISSUE #279: Unable to form a cluster when all nodes are up and running
### Summary 
 The easiest way to reproduce in a 4-nodes configuration is to scale the stateful set to 2 nodes, then to 3 and, after a minute, to 4 nodes. In the majority cases, it will result in a split condition detected by ccow-daemon while all pods are up and running.

There are at least 2 groups of reasons.

1. The ccow-daemon isn't registered in the "nexenta-ccow"  CPG. The corosync itself is able to build consensus and provides services. However, the `corosync-cpgtool` is showing that the current instance hasn't joined the "nexent-ccow" group.
The problem might be related to a corocync configuration reload request made by a script when a pod is starting. The problem is pretty difficult to reproduce. It disappears after sending SIGUSR1 to the ccow-daemon

2. The corosync's membership protocol is failing to build consensus when pods are restarted in random order. Need to mention that present configuration uses POD names instead of IPs and thus requires a DNS lookup after each POD restart. At the moment, there is the only way to make corosync to re-build node IPs table: to send it a configuration reload request. However, it seems a reload request affects internal parts of corosync. Presumably due to a delay introduced by DNS lookup which may take up to 1 minute and repeats several times. The following anomalies were detected by analyzing corosync's logs:

- The memership protocol is unable to build a consensus when all nodes are online and participates in negotiations. We see all nodes sending JOIN messages with correct and equal `proc_set` and `fail_set` fields, but COMMIT token isn't sent by the representative, instead JOIN messages keep emitted by negotiators. It continues until a node adds another one to its "fail_set" and the COMMIT token is emitted. Sometimes the JOIN process keeps until pod restart. Reproduced with both knet and udpu drivers.

- A corosync instance is starting and sending a JOIN messages. However, other nodes aren't receiving the message and the node stays alone. Often reproducible with knet driver.

- The knet driver is unable to ping other nodes even if they are online. Rear situation, reproduced only once.

### Deployment type

- [ ] Kubernetes with Rook Operator
- [ ] Docker Compose

- [ ] Embedded with less than 8GB per EdgeFS Target or Daemon container
- [ ] Optimal resource allocation with more then 4 CPU and 8GB per EdgeFS Target or Daemon container

### Kubernetes cluster

- [ ] RedHat 7.x
- [ ] Ubuntu 16.04.x
- [ ] Ubuntu 18.04.x
- [ ] Other

### PR DESCRIPTION
**Pull request by SergeyKaydalov**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.
- [ ] Change modifies documentation or comments only.

#### Description
- Added a coroync-pow-watcher tool which modifies a corosync.conf when a new target pod is joining the cluster
- Using a shared GOPATH
-  the efscli and coroync-pow-watcher  are built as go modules

#### Which issue this PR fixes

fixes #279

### CODE REVIEW

File: src/corosync-pod-watcher/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/Nexenta/edgefs/src/corosync-pod-watcher
```
Comment (dyusupov) [inline src/corosync-pod-watcher/go.mod]: I do not see go.sum file - isn't that need to be added as well?

File: src/.gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 .ccow-build
 .ccow-configure
+src
```
Comment (dyusupov) [inline src/.gitignore]: and bin

File: src/corosync-pod-watcher/go.mod (hunk 1 above)
Comment (SergeyKaydalov) [reply src/corosync-pod-watcher/go.mod]: it's autogenerated on build

File: src/.gitignore (hunk 2 above)
Comment (SergeyKaydalov) [reply src/.gitignore]: adding..

File: src/corosync-pod-watcher/go.mod (hunk 1 above)
Comment (SergeyKaydalov) [reply src/corosync-pod-watcher/go.mod]: Added go.sum

**Comment by dyusupov** [comment]
:star: PR built and available in a preview environment **nexenta-edgefs-pr-297**

**Comment by dyusupov** [comment]
:star: PR built and available in a preview environment **nexenta-edgefs-pr-297**

**Pull request merged**